source/backend/api/Services/NoteService.cs

```diff
@@ -165,27 +180,37 @@ public bool DeleteNote(NoteType type, long noteId, bool commitTransaction = true
         /// Get notes by note type.
         /// </summary>
         /// <param name="type">Note type to determine the type of notes to return.</param>
-        /// <param name="entityId">Entity Id to determine the entity to which notes belongs to.</param>
+        /// <param name="parentId">Entity Id to determine the entity to which notes belongs to.</param>
         /// <returns></returns>
-        public IEnumerable<PimsNote> GetNotes(NoteType type, long entityId)
+        public IEnumerable<PimsNote> GetNotes(NoteType type, long parentId)
         {
-            Logger.LogInformation("Getting all notes with type {Type} and parent id {EntityId}", type, entityId);
+            Logger.LogInformation("Getting all notes with type {Type} and parent id {EntityId}", type, parentId);
             User.ThrowIfNotAuthorized(Permissions.NoteView);
 
-            List<PimsNote> notes = type switch
+            IList<PimsNote> notes = type switch
             {
-                NoteType.Acquisition_File => _entityNoteRepository.GetAllAcquisitionNotesById(entityId).ToList(),
-                NoteType.Disposition_File => _entityNoteRepository.GetAllDispositionNotesById(entityId).ToList(),
-                NoteType.Project => _entityNoteRepository.GetAllProjectNotesById(entityId).ToList(),
-                NoteType.Lease_File => _entityNoteRepository.GetAllLeaseNotesById(entityId).ToList(),
-                NoteType.Research_File => _entityNoteRepository.GetAllResearchNotesById(entityId).ToList(),
-                NoteType.Management_File => _entityNoteRepository.GetAllManagementNotesById(entityId).ToList(),
+                NoteType.Acquisition_File => _acquisitionFileNoteRepository.GetAllByParentId(parentId),
+                NoteType.Disposition_File => _dispositionFileNoteRepository.GetAllByParentId(parentId),
+                NoteType.Project => _projectNoteRepository.GetAllByParentId(parentId),
+                NoteType.Lease_File => _leaseNoteRepository.GetAllByParentId(parentId),
+                NoteType.Research_File => _researchFileNoteRepository.GetAllByParentId(parentId),
+                NoteType.Management_File => _managementFileNoteRepository.GetAllByParentId(parentId),
+                NoteType.Property => _propertyNoteRepository.GetAllByParentId(parentId),
                 _ => new List<PimsNote>()
             };
 
             return notes;
         }
 
+        private static T AddNoteRelationship<T>(T noteRelationship, INoteRelationshipRepository<T> noteRelationshipRepository)
+            where T : PimsNoteRelationship, new()
+        {
+            T newEntity = noteRelationshipRepository.AddNoteRelationship(noteRelationship);
+            noteRelationshipRepository.CommitTransaction();
+
+            return newEntity;
+        }
+
```

Comment:
Generic function to add note-relationships (property-note, acquisition-note, etc)

Reply:
nit: the new() might not be necessary